### PR TITLE
feat: [CDS-80768]: edit/update flow of gitx infra

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
### Summary

- Integrate the get-flow of the remote infrastructure entity on editing infrastructure .
- Introduced new queryparams for infrastructure entity to take reference for branch change and maintain existing flow.
- Tested inline flow 
- Tested remote flow and checked if on editing existing remote infrastructure, infrastructure is updated properly without error.
- if infrastructure has been deleted from github, no entity found component is diplayed.

#### Screenshots

https://github.com/harness/harness-core-ui/assets/98508399/80021940-634e-4ee7-8d37-33193b1a9f33

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx [hunk 1]
```diff
@@ -212,10 +214,31 @@ export function GitSyncForm<T extends GitSyncFormFields = GitSyncFormFields>(
...
+    entityType === EntityType.Infrastructure
+      ? {
+          connectorRef: infraConnectorRef,
+          branch: infraBranch,
+          repoName: infraRepoName
+        }
+      : {
+          connectorRef: connectorRefFromUrl,
+          branch: branchFromUrl,
+          repoName: repoNameFromUrl
```
Comment (sonaligoyal-harness) [inline src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx:240]: Note: this is done so that when infrastructure is created within remote environment, initialValues in create modal for these fields are not shown on UI for infra.

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx [hunk 2]
```diff
@@ -817,14 +862,20 @@ function BootstrapDeployInfraDefinition(
             <>
               <Card className={css.infraCardSection}>
                 <Formik<Partial<CombinedInfrastructureDefinationResponse> & GitSyncFormFields>
-                  initialValues={formValues}
+                  initialValues={
+                    !isEmpty(selectedInfrastructure) && infraStoreType === 'REMOTE'
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx:866]: will get created on every render, can be memoised

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfraDefinitionDetailsDrawer/InfraDefinitionDetailsDrawer.tsx [hunk 3]
```diff
@@ -45,6 +62,9 @@ interface Props {
   handleInfrastructureUpdate?: (updatedInfrastructure: InfrastructureConfig) => void
   updatedInfra?: InfrastructureConfig
   isSingleEnv?: boolean
+  infrastructureResponse?: InfrastructureResponseDTO
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfraDefinitionDetailsDrawer/InfraDefinitionDetailsDrawer.tsx:65]: instead of passing three props, check if we can pass the return value of api hook directly and then destructure it here in child to get all three

so rather than passing 3 props - this can be done in a single prop

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.module.scss [hunk 4]
```diff
@@ -171,6 +171,16 @@
   width: 450px;
 }
 
+.gitRemoteDetailsWrapper {
+  display: inline-flex;
+  align-items: center;
+  padding-left: 12px;
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.module.scss:177]: can we use css standard variables here ?

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.module.scss [hunk 5]
```diff
@@ -171,6 +171,16 @@
...
+  border: 1px solid var(--grey-200) !important;
+  border-radius: var(--spacing-2);
+  height: 32px;
+  margin-left: 14px;
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.module.scss:181]: https://github.com/harness/harness-core-ui/pull/18471/files#r1369260738

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx [hunk 6]
```diff
@@ -66,6 +72,28 @@ export default function InfrastructureDefinition({ isEnvPage }: { isEnvPage: boo
...
+    data: infrastructureData,
+    loading: infrastructureLoading,
+    error: infrastructureError
+  } = useGetInfrastructure({
+    queryParams: {
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier,
+      environmentIdentifier,
+      ...(infraStoreType === 'REMOTE'
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:85]: To increase readability, lets extract it in a separate variable above and then use it here (lets say const gitParams)

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx [hunk 7]
```diff
@@ -138,7 +166,10 @@ export default function InfrastructureDefinition({ isEnvPage }: { isEnvPage: boo
             openUnsavedChangesDiffModal={openUnsavedChangesDiffModal}
             handleInfrastructureUpdate={handleInfrastructureUpdate}
             updatedInfra={updatedInfrastructure}
+            infrastructureResponse={infrastructureData?.data?.infrastructure}
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:169]: Instead of three props (infrastructureResponse , infrastructureError, infrastructureLoading), lets use the return value of "useGetInfrastructure" directly and pass it to child (can be named infrastructureFetchDetails)

and then in child component -> 

const  {infrastructureResponse ,infrastructureError , infrastructureLoading } =  infrastructureFetchDetails

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx [hunk 8]
```diff
@@ -29,16 +29,18 @@ import produce from 'immer'
...
 import { usePipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
 import { useStrings } from 'framework/strings'
 import { StageErrorContext } from '@pipeline/context/StageErrorContext'
 import {
   CustomDeploymentConnectorNGVariable,
   DeploymentStageConfig,
+  EntityGitDetails,
   InfrastructureConfig,
   InfrastructureDefinitionConfig,
   InfrastructureResponseDTO,
```
Comment (samarth-sharma31) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx:42]: Although it is unrelated, but please test reference by section of remote infrastructure

Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx:42]: There is no change as of now in that section.
But will test it by keeping remote env/infra cases into consideration.

**Review by samarth-sharma31 — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx [hunk 9]
```diff
@@ -794,6 +833,22 @@ function BootstrapDeployInfraDefinition(
...
+      filePath: infrastructureResponse?.entityGitDetails?.filePath,
+      storeType: infrastructureResponse?.storeType
+    }
+
+    if (!isEmpty(selectedInfrastructure) && infraStoreType === 'REMOTE') {
+      return { ...formValues, ...initialGitFormValues }
+    } else {
+      return formValues
+    }
+  }, [infraStoreType, selectedInfrastructure])
```
[the remark this answers is not in the document]
Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx:850]: Will update in next PR.

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx [hunk 10]
```diff
@@ -212,10 +214,31 @@ export function GitSyncForm<T extends GitSyncFormFields = GitSyncFormFields>(
     filePathPrefix,
     skipDefaultConnectorSetting = false,
     skipBranch = false,
-    supportNewBranch = false
+    supportNewBranch = false,
+    entityType
   } = props
   const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
-  const { branch, connectorRef, repoName } = useQueryParams<GitQueryParams>()
+  const {
```
Comment (sunnykesh-harness) [inline src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx:221]: Let us revert this as discussed.

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx (hunk 10 above)
Comment (sonaligoyal-harness) [reply src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx]: Done

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx [hunk 11]
```diff
@@ -215,7 +215,7 @@ export function GitSyncForm<T extends GitSyncFormFields = GitSyncFormFields>(
     supportNewBranch = false
   } = props
   const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
-  const { branch, connectorRef, repoName } = useQueryParams<GitQueryParams>()
+  const { connectorRef, branch, repoName } = useQueryParams<GitQueryParams>()
```
Comment (sunnykesh-harness) [inline src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx:218]: [Nit] Unnecessary changes.

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx [hunk 12]
```diff
@@ -794,6 +836,26 @@ function BootstrapDeployInfraDefinition(
...
+          connectorRef,
+          repo: repoName
+        }
+      : {
+          connectorRef: infrastructureResponse?.connectorRef,
+          repo: infrastructureResponse?.entityGitDetails?.repoName,
+          filePath: infrastructureResponse?.entityGitDetails?.filePath,
+          storeType: infrastructureResponse?.storeType
+        }
+    if (isEmpty(selectedInfrastructure) || infraStoreType === 'REMOTE') {
```
Comment (sunnykesh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/BootstrapDeployInfraDefinition.tsx:852]: Plz make sure, these condition do not affect existing Inline flows.

**Review by sunnykesh-harness — APPROVED** [review]
Bugs were identifies for remote infra which should be fixed in follow up PR.

**Review by vikrant-harness — APPROVED** [review]
